src/app/components/LoadMoreProducts.tsx

```
@@ -28,16 +28,32 @@ export const LoadMoreProducts = memo(
       handleSkipVal,
     } = useProducts();
+
     const [products, setProducts] = useState<
-      Product[] | undefined
+      Product[] | FullProduct[]
     >(
       category === 'all'
         ? [
             ...hardCodedProducts,
-            ...initialProducts,
+            ...initialProducts?.map((p) => ({
```

Biome (1.9.4)

[error] Unsafe usage of optional chaining.

If it short-circuits with 'undefined' the evaluation will throw TypeError here:


(lint/correctness/noUnsafeOptionalChaining)